inc/components/class-head.php

```diff
@@ -170,4 +180,154 @@ protected function add_tag( $tag, $attributes = [] ) {
 			true
 		);
 	}
+
+	/**
+	 * Apply basic meta tags.
+	 *
+	 * @param  integer $post_id Post ID.
+	 */
+	public function apply_meta( $post_id ) {
+
+		// Meta description.
+		$meta_description = $this->get_meta_description( $post_id );
+		if ( ! empty( $meta_description ) ) {
+			$this->add_meta( 'description', esc_attr( $meta_description ) );
+		}
+
+		// Meta keywords.
+		$meta_keywords = (string) get_post_meta( $post_id, '_meta_keywords', true );
+		if ( ! empty( $meta_keywords ) ) {
+			$this->add_meta( 'keywords', esc_attr( $meta_keywords ) );
+		}
+	}
+
+	/**
+	 * Apply social meta tags.
+	 *
+	 * @param  integer $post_id Post ID.
+	 */
+	public function apply_social_meta( $post_id ) {
+
+		// Open graph meta.
+		$this->add_meta( 'og:url', get_the_permalink( $post_id ) );
+		$this->add_meta( 'og:type', 'article' );
+		$this->add_meta( 'og:title', $this->get_social_title( $post_id ) );
+		$this->add_meta( 'og:description', $this->get_social_description( $post_id ) );
+
+		// Optional meta.
+		$image_url = $this->get_social_image_url( $post_id, [ 'resize' => '1200,630' ] );
+		if ( ! empty( $image_url ) ) {
+			$this->add_meta( 'og:image', $image_url );
+		}
+
+		// Twitter specific meta.
+		$this->add_meta( 'twitter:card', 'summary' );
+	}
+
+	/**
+	 * Get the meta title for a post.
+	 *
+	 * @param  integer $post_id Post ID.
+	 * @return string Meta title.
+	 */
+	public function get_meta_title( $post_id ) : string {
+
+		// Return WP-SEO title.
+		$meta_title = (string) get_post_meta( $post_id, '_meta_title', true );
+		if ( ! empty( $meta_title ) ) {
+			return $meta_title;
+		}
+
+		return (string) get_the_title( $post_id );
+	}
+
+	/**
+	 * Get the social title for a post.
+	 *
+	 * @param  integer $post_id Post ID.
+	 * @return string Social title.
+	 */
+	public function get_social_title( $post_id ) : string {
+
+		// Use social title, or fallback to meta title.
+		$social_title = get_post_meta( $post_id, 'social_title', true );
+		if ( ! empty( $social_title ) ) {
+			return $social_title;
+		}
+
+		return (string) $this->get_meta_title( $post_id );
+	}
+
+	/**
+	 * Get the meta descrption for a post.
+	 *
+	 * @param  integer $post_id Post ID.
+	 * @return string Meta description.
+	 */
+	public function get_meta_description( $post_id ) : string {
+
+		// Return WP-SEO description.
+		$meta_description = (string) get_post_meta( $post_id, '_meta_description', true );
+		if ( ! empty( $meta_description ) ) {
+			return $meta_description;
+		}
+
+		// Modify global state.
```

Comment:
The following should accomplish the same without modifying global state and having to set the temp variable `apply_filters( 'get_the_excerpt', get_post_field( 'post_excerpt', $post_id ) )`

Reply:
I had a long discussion with @mboynes about this and this was his recommended approach. He suggested avoiding the filter all together.